appinventor/appengine/src/com/google/appinventor/client/explorer/commands/DeleteFileCommand.java

```diff
@@ -36,6 +69,195 @@ public boolean willCallExecuteNextCommand() {
 
   @Override
   public void execute(final ProjectNode node) {
+	  if (node instanceof YoungAndroidSourceNode) {
+  		new NewFormDialog((YoungAndroidSourceNode) node).center();
+  		
+	  } 
+	  else {
+  		  executionFailedOrCanceled();
+  		  throw new IllegalArgumentException("node must be a YoungAndroidProjectNode");
+	  }
+  }
+
+  /**
+   * Shows a confirmation dialog.
+   *
+   * @return {@code true} if the delete file command should be executed or
+   *         {@code false} if it should be canceled
+   */
+  protected boolean deleteConfirmation() {
+    return Window.confirm(MESSAGES.reallyDeleteFile());
+  }
+  
+  private class NewFormDialog extends DialogBox {
+    // UI elements
+    private final LabeledTextBox newNameTextBox;
+	private final CheckBox cb = new CheckBox(MESSAGES.checkBoxButton());
+    private final Set<String> otherFormNames;
+
+    NewFormDialog(final YoungAndroidSourceNode projectRootNode) {
+      super(false, true);
+      setStylePrimaryName("ode-DialogBox");
+      VerticalPanel contentPanel = new VerticalPanel();
+	  setText(MESSAGES.removeFormButton());
+      final String prefix = "Screen";
+      final String abc;
+      final int prefixLength = prefix.length();
+      int highIndex = 0;
+      // Collect the existing form names so we can prevent duplicate form names.
+      otherFormNames = new HashSet<String>();
+
+      /*for (ProjectNode source : projectRootNode.getAllSourceNodes()) {
+        if (source instanceof YoungAndroidFormNode) {
+          String formName = ((YoungAndroidFormNode) source).getFormName();
+          otherFormNames.add(formName);
+
+          if (formName.startsWith(prefix)) {
+            try {
+              highIndex = Math.max(highIndex, Integer.parseInt(formName.substring(prefixLength)));
+            } catch (NumberFormatException e) {
+              continue;
+            }
+          }
+        }
+      }*/
+
+      String defaultFormName = prefix + (highIndex + 1);
+
+      newNameTextBox = new LabeledTextBox(MESSAGES.formNameLabel());
+      newNameTextBox.setText(defaultFormName);
+      newNameTextBox.getTextBox().addKeyUpHandler(new KeyUpHandler() {
+        @Override
+        public void onKeyUp(KeyUpEvent event) {
+          int keyCode = event.getNativeKeyCode();
+          if (keyCode == KeyCodes.KEY_ENTER) {
+            //handleOkClick(projectRootNode);
```

Comment:
Is there a reason we're checking for the enter key here? The click call is commented out.

Reply:
Removed KeyUp Handler